src/main/java/duck/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duck;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * Handles saving and loading tasks from a file.
+ */
+public class Storage {
+    protected String filePath;
+
+    /**
+     * Constructs a Storage instance with the specified file path.
+     *
+     * @param filePath The path of the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves the task list to a file.
+     *
+     * @param filePath The file path to save the tasks.
+     * @param list The list of tasks to save.
+     */
+    public void save(String filePath, TaskList list) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            for (Task task : list.getAllTasks()) {
+                writer.write(taskToString(task));
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.err.println("Error saving tasks to file: " + e.getMessage());
+            System.out.println("_________________________________________________");
+        }
+    }
+
+    /**
+     * Loads tasks from the storage file.
+     *
+     * @return A list of tasks retrieved from the file.
+     * @throws IOException If an error occurs while reading the file.
+     */
+    public ArrayList<Task> load() throws IOException {
+        ArrayList<Task> list = new ArrayList<Task>();
```

Comment:
nit: Should diamond operator be used?

```suggestion
        ArrayList<Task> list = new ArrayList<>();
```